app/code/Magento/Checkout/Model/GuestPaymentInformationManagement.php

```diff
@@ -141,7 +157,7 @@ public function getPaymentInformation($cartId)
     private function getLogger()
     {
         if (!$this->logger) {
-            $this->logger = \Magento\Framework\App\ObjectManager::getInstance()->get(\Psr\Log\LoggerInterface::class);
+            $this->logger = ObjectManager::getInstance()->get(\Psr\Log\LoggerInterface::class);
```

Comment:
How about we take this a step further? Since we are changing the `__construct` anyway why not optionally inject the logger also and then we can remove this `getLogger` method completely.